exonum-java-binding/core/pom.xml

```diff
@@ -421,5 +441,31 @@
         </plugins>
       </build>
     </profile>
+
+    <profile>
+      <id>app-packaging-macos</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-assembly-plugin</artifactId>
+            <configuration>
+              <descriptors>
+                <descriptor>app_assembly_macos.xml</descriptor>
+              </descriptors>
+            </configuration>
+            <executions>
+              <execution>
+                <id>make-assembly</id>
```

Comment:
As it's a user-defined id, I'd give both a name explaining this step (make-app-assembly or package-app-macos/linux)